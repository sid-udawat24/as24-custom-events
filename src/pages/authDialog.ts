export const AuthDialogTypeChangeEvent = 'AUTH_DIALOG_TYPE_CHANGE_EVENT'
export const AuthDialogFragmentReadyEvent = 'AUTH_DIALOG_FRAGMENT_READY_EVENT'

export type AuthDialogFragmentReady = {}
export type AuthDialogTypeChange = {
  /**
   * clientId - OKTA app id, see https://docs.services.as24.tech/autoscout24/products/identity/howto/#register-a-new-client
   */
  clientId: string
  /**
   * final redirect to go to after successful sign/signup. If not defined, the current window.location.href will be used.
   * IMPORTANT: If the value is not autoscout24 domain, no redirection will happens.
   */
  redirectUrl?: string
  /**
   * The touchpoint of signin/signup attempt, i.e. 'pricehistory', 'pricehistory_icon'
   * Used to track opening event in GA.
   */
  tp?: string
  /**
   * Override existing texts on social-, login and registration view within the overlay
   */
  overrideText?: {
    // "Sign In"
    signInTitle: string
    // "Welcome!"
    signInTeaserTitle: string
    // "Log in and keep all your searches in one place."
    signInTeaserDescription: string
    // "Sign In"
    socialTitle: string
    // "Log in and keep all your searches in one place."
    socialDescription: string
    // "Sign up"
    signUpTitle: string
    // "Sign up and keep all your searches in one place."
    signUpDescription: string
  }
}

declare global {
  interface DocumentEventMap {
    [AuthDialogTypeChangeEvent]: CustomEvent<AuthDialogTypeChange>
    [AuthDialogFragmentReadyEvent]: CustomEvent<AuthDialogFragmentReady>
  }
}
