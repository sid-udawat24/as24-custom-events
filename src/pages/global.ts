import * as ListPage from './listpage'

// Events that are present across all pages

/**
 * Privacy Consent Service UI events to show overlay
 */
export const PcsUiShowOverlayEvent = 'PCS_UI_SHOW_OVERLAY'
export type PcsUiShowOverlayEventPayload = {
  type: typeof PcsUiShowOverlayEvent
  timeout: number
}

/**
 * Privacy Consent Service UI events to show DOI OptIn overlay
 */
export const PcsUiShowDoiOptinOverlayEvent = 'PCS_UI_SHOW_DOI_OPTIN_OVERLAY'
export type PcsUiShowDoiOptinOverlayEventPayload = {
  type: typeof PcsUiShowDoiOptinOverlayEvent
  timeout: number
}

/**
 * Watchlist event
 */
export const WatchlistEvent = 'AS24_WATCHLIST_EVENT'
export const WatchlistEventCountUpdate = 'WATCHLIST_COUNT_UPDATE'
export type WatchlistEventCountUpdate = {
  type: typeof WatchlistEventCountUpdate
  value: number
}
export type WatchlistEvent = WatchlistEventCountUpdate

/**
 * Indicates that the favorites client finished the initialization
 * and is ready to accept commands.
 */
export const FavoritesReady = 'FAVORITES:READY'
export type FavoritesReady = undefined

export const FavoriteAddedEvent = 'WATCHLIST:FAVORITE_ADDED'
export const FavoriteRemovedEvent = 'WATCHLIST:FAVORITE_REMOVED'

export type FavoriteAddedEventPayload = {
  type: typeof FavoriteAddedEvent
  value: string
}
export type FavoriteRemovedEventPayload = {
  type: typeof FavoriteRemovedEvent
  value: string
}

// inform compiler of the as24 custom events
declare global {
  interface DocumentEventMap {
    [WatchlistEvent]: CustomEvent<WatchlistEvent>
    [FavoritesReady]: CustomEvent<FavoritesReady>
    [FavoriteAddedEvent]: CustomEvent<FavoriteAddedEventPayload>
    [FavoriteRemovedEvent]: CustomEvent<FavoriteRemovedEventPayload>
    [PcsUiShowOverlayEvent]: CustomEvent<PcsUiShowOverlayEventPayload>
    [PcsUiShowDoiOptinOverlayEvent]: CustomEvent<PcsUiShowDoiOptinOverlayEventPayload>
  }
}
