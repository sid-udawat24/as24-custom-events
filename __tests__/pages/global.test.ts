import * as p from '../../src/pages/global'

describe('global events', () => {
  const eventNames: string[] = [
    p.WatchlistEvent,
    p.FavoritesReady,
    p.FavoriteAddedEvent,
    p.FavoriteRemovedEvent,
    p.PcsUiShowOverlayEvent,
    p.PcsUiShowDoiOptinOverlayEvent
  ]

  test.each(eventNames)('event names must never change', eventName => {
    expect(eventName).toMatchSnapshot()
  })

  test('type definitions are inferred correctly by compiler', () => {
    document.addEventListener(p.WatchlistEvent, e => e.detail.value as number)
    document.addEventListener(p.FavoritesReady, e => e.detail as undefined)
    document.addEventListener(p.FavoriteAddedEvent, e => e.detail.value as string)
    document.addEventListener(p.FavoriteRemovedEvent, e => e.detail.value as string)
    document.addEventListener(p.PcsUiShowOverlayEvent, e => e.detail.timeout as number)
    document.addEventListener(p.PcsUiShowDoiOptinOverlayEvent, e => e.detail.timeout as number)
    expect(true).toBeTruthy() // dummy assertion since compiler check is the real test
  })
})
