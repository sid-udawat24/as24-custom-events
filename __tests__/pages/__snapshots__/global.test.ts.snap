// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global events event names must never change 1`] = `"AS24_WATCHLIST_EVENT"`;

exports[`global events event names must never change 2`] = `"FAVORITES:READY"`;

exports[`global events event names must never change 3`] = `"WATCHLIST:FAVORITE_ADDED"`;

exports[`global events event names must never change 4`] = `"WATCHLIST:FAVORITE_REMOVED"`;

exports[`global events event names must never change 5`] = `"PCS_UI_SHOW_OVERLAY"`;

exports[`global events event names must never change 6`] = `"PCS_UI_SHOW_DOI_OPTIN_OVERLAY"`;
